.github/workflows/release-chart.yaml

```diff
@@ -31,3 +31,22 @@ jobs:
         env:
           CR_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
           CR_RELEASE_NAME_TEMPLATE: "metrics-server-helm-chart-{{ .Version }}"
+
+      - name: Get chart app version
+        id: chart_app_version
+        uses: mikefarah/yq@v4.12.2
+        with:
+          cmd: yq eval '.appVersion' './charts/metrics-server/Chart.yaml'
```

Comment:
Do we have any guarantees that the chart will always be based on the latest releases of metrics-server?

How about we just copy the latest manifests that are hosted on`https://github.com/kubernetes-sigs/metrics-server/releases/latest/`? This way we would not have to bother with the chart lagging behind and it would make sure that we always have the latest manifest pushed since the chart will not update them.

Reply:
@dgrisonnet as a first principal we shouldn't attach a manifest to a release that it isn't relevant to, so this is the correct behaviour. I doubt there would ever be a backported chart release, and if for some unexpected reason there was it could be coordinated correctly with an active release chart to leave latest set correctly.

Long term I assume that the use of `latest` will be stopped so this hack can be deprecated? Maybe pushing the latest manifests to the `gh-pages` branch would give a static resource and the CI could only do this on a newer release?

Reply:
Yeah, I agree that we shouldn't attach manifests to irrelevant releases, but if we were to base the copy of the assets on the `latest` release rather than the version of `metrics-server`, it would make sure that the assets are always in sync with the latest release of metrics-server since only official releases can update the assets and then helm releases will propagate the latest version.
However, if we base it on the version of metrics-server used in the chart, it will depend on the chart version and if you were to release a version of the chart that uses metrics-server v0.5.0 after release v0.6.0, then we would end up with an incorrect version of the assets.

Reply:
@dgrisonnet the only way to make sure that the manifests attached aren't from a different releases is to use the chart `appVersion` to find the correct manifests for the version represented by the chart, this also makes the behaviour deterministic.

Everything else is just process; so if there needs to be a rule requiring all Helm chart releases be from the latest Metrics Server release that can be defined. Alternatively a flexible approach could be taken where if anything is released to leave a non-latest release in the latest position a dummy release could be added to carry the manifests (either a chart patch or a custom release and tag).

Reply:
My point is that the manifests shouldn't be tied to the chart. In my opinion, this job should only copy the manifests from the latest release and not have the version of the manifests depend on the chart. If we were to copy from latest we would have the following flow:
- release v0.5.0: push v0.5.0 manifests
- metrics-server-helm-chart-3.5.0: copy /releases/latest manifests which correspond to v0.5.0
- metrics-server-helm-chart-3.5.1: copy /releases/latest manifests from chart 3.5.0 which correspond to v0.5.0
- release v0.6.0: push v0.6.0 manifests
- metrics-server-helm-chart-3.6.0: copy /releases/latest manifests which correspond to v0.6.0
- metrics-server-helm-chart-3.6.1: copy /releases/latest manifests from chart 3.6.0 which correspond to v0.6.0

This way the manifests attached to helm chart releases will always be based on the latest official release, no matter which version of metrics-server the chart is using.

Reply:
@dgrisonnet the flow you've shown would result in exactly the same output as using the `appVersion` workflow in this PR. By using the `appVersion` workflow the behaviour would be deterministic and manifests would never be associated with the wrong release. By using `/releases/latest` the process is non-deterministic and could not only result in incorrect manifests being attached to a release, but could also be broken to the point of blocking and needing manual resolution.